validations/src/validations.rs

```diff
@@ -247,21 +247,26 @@ pub fn validate_rad_request(
     }
 
     for path in retrieval_paths {
-        // Regarding WIP-0019 activation:
-        // Before -> Only RadType enum 0 position is valid
-        // After -> RadType::Unknown are invalid
-        if (!active_wips.wip0019() && path.kind != RADType::Unknown)
-            || (active_wips.wip0019() && path.kind == RADType::Unknown)
-        {
-            return Err(DataRequestError::InvalidRadType.into());
-        }
+        if active_wips.wip0020() {
+            path.check_fields()?;
+            unpack_radon_script(path.script.as_slice())?;
+        } else {
+            // Regarding WIP-0019 activation:
+            // Before -> Only RadType enum 0 position is valid
+            // After -> RadType::Unknown are invalid
+            if (!active_wips.wip0019() && path.kind != RADType::Unknown)
+                || (active_wips.wip0019() && path.kind == RADType::Unknown)
+            {
+                return Err(DataRequestError::InvalidRadType.into());
+            }
 
-        let is_rng = active_wips.wip0019() && path.kind == RADType::Rng;
-        // If the sources are empty the data request is set as invalid
-        if path.url.is_empty() && !is_rng {
-            return Err(DataRequestError::NoRetrievalSources.into());
+            let is_rng = active_wips.wip0019() && path.kind == RADType::Rng;
+            // If the sources are empty the data request is set as invalid
+            if path.url.is_empty() && !is_rng {
+                return Err(DataRequestError::NoRetrievalSources.into());
+            }
+            unpack_radon_script(path.script.as_slice())?;
```

Comment:
could be refactored as an only if-elseif-else clause?

Reply:
I tried to avoid touching the "legacy code". Is it better after the latest change?